org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/conformance/RawTypeConformanceComputer.java

```diff
@@ -649,6 +652,25 @@ LightweightMergedBoundTypeArgument getMergeResult(UnboundTypeReference left) {
 		return flags;
 	}
 	
+	private boolean isConstrainedRecursiveHintCheck(List<LightweightBoundTypeArgument> leftHints, LightweightTypeReference right) {
+		boolean hasConstraints = false;
+		for(LightweightBoundTypeArgument leftHint: leftHints) {
```

Comment:
how are these left hints sorted?

Reply:
They are returned in the order of their appearance. The first hints are usually deduced from the type parameter declarations.